dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ViewContentletAction.java

```diff
@@ -1,165 +1,199 @@
+      final RenderResponse res) throws Exception {
+
+    Logger.debug(ViewContentletAction.class, "Running ViewContentletsAction!!!!");
+
+    try {
+      // gets the user
+      final User user = _getUser(req);
+      _viewContentlets(req, user);
+      return mapping.findForward("portlet.ext.contentlet.view_contentlets");
+
+    } catch (final Exception e) {
+      req.setAttribute(PageContext.EXCEPTION, e);
+      return mapping.findForward(Constants.COMMON_ERROR);
+    }
+  }
+
+  private List<BaseContentType> resolveBaseTypes(HttpServletRequest request) {
+    Map<String, String> initParams = (Map<String, String>) request.getAttribute("initParams");
+    String baseTypesRaw = (initParams.get("baseTypes") != null) ? initParams.get("baseTypes") : request.getParameter("baseType");
+    if (!UtilMethods.isSet(baseTypesRaw)) {
+      return ImmutableList.of();
+    }
+    String[] baseTypes = baseTypesRaw.trim().split(",");
+    List<BaseContentType> baseTypeList = new ArrayList<>();
+    for (String type : baseTypes) {
+      if (UtilMethods.isSet(type) && UtilMethods.isNumeric(type)) {
+        baseTypeList.add(BaseContentType.getBaseContentType(Integer.parseInt(type)));
+      } else {
+        baseTypeList.add(BaseContentType.getBaseContentType(type));
+      }
+    }
+    return baseTypeList;
+  }
+
+  List<ContentType> resolveContentTypes(HttpServletRequest request) throws DotDataException, DotSecurityException {
+    Map<String, String> initParams = (Map<String, String>) request.getAttribute("initParams");
+
+    ContentTypeAPI contentTypeApi = APILocator.getContentTypeAPI(PortalUtil.getUser(request));
+    final List<ContentType> contentTypesList = new ArrayList<>();
+    List<BaseContentType> baseTypes = resolveBaseTypes(request);
+
+    if (baseTypes.size() > 0) {
+      for (BaseContentType type : baseTypes) {
+        contentTypesList.addAll(contentTypeApi.findByType(type));
+      }
+      request.setAttribute("contentTypesJs", buildJsArray(contentTypesList));
+      return contentTypesList;
+    }
+
+    String contentTypesRaw = initParams.getOrDefault("contentTypes", request.getParameter("structure_id"));
+    if (UtilMethods.isSet(contentTypesRaw)) {
+      String[] contentTypes = contentTypesRaw.trim().split(",");
+      List<ContentType> contentTypeList = new ArrayList<>();
+      for (String type : contentTypes) {
+          ContentType contentType = Try.of(() -> APILocator.getContentTypeAPI(PortalUtil.getUser(request)).find(type)).getOrNull();
+          if (contentType != null) {
+            contentTypeList.add(contentType);
+          }
+      }
+      request.setAttribute("contentTypesJs", buildJsArray(contentTypesList));
+      return contentTypeList;
+    }
+
+    return contentTypeApi.findAll();
+
+  }
+
+  private String buildJsArray(List<ContentType> types) {
+    String typeStr = String.join(",", types.stream().map(t ->  t.id() ).collect(Collectors.toList()));
```

Comment:
Return the String.join directly without creating the new String

Reply:
done